### PR TITLE
Msw 113961399 fix activity table

### PR DESCRIPTION
**Pull request by michaelwehrley**

### CODE REVIEW

File: app/models/group.rb [hunk 1]
```diff
@@ -55,8 +57,10 @@ def activities_past_by_week(week_number)
         .participant
         .activities
         .in_the_past
-        .where(activities[:created_at].gteq(week_start_day(week_number)))
-        .where(activities[:created_at].lt(week_end_day(week_number)))
+        .where(activities[:start_time]
```
Comment (michaelwehrley) [inline app/models/group.rb]: note change to `start_time`

File: app/models/group.rb [hunk 2]
```diff
@@ -68,8 +72,10 @@ def activities_future_by_week(week_number)
         .participant
         .activities
         .unscheduled_or_in_the_future
-        .where(activities[:created_at].gteq(week_start_day(week_number)))
-        .where(activities[:created_at].lt(week_end_day(week_number)))
+        .where(activities[:start_time]
```
Comment (michaelwehrley) [inline app/models/group.rb]: note change to `start_time`

File: app/models/group.rb [hunk 3]
```diff
@@ -42,8 +42,10 @@ def thoughts_by_week(week_number)
       membership
         .participant
         .thoughts
-        .where(thoughts[:created_at].gteq(week_start_day(week_number)))
-        .where(thoughts[:created_at].lt(week_end_day(week_number)))
+        .where(thoughts[:created_at]
+          .gteq(membership.week_start_day(week_number)))
```
Comment (michaelwehrley) [inline app/models/group.rb]: note: `week_start_day` is called from `membership` - no longer the `group`

File: app/models/group.rb [hunk 4]
```diff
@@ -55,8 +57,10 @@ def activities_past_by_week(week_number)
         .participant
         .activities
         .in_the_past
-        .where(activities[:created_at].gteq(week_start_day(week_number)))
-        .where(activities[:created_at].lt(week_end_day(week_number)))
+        .where(activities[:start_time]
+          .gteq(membership.week_start_day(week_number)))
```
Comment (michaelwehrley) [inline app/models/group.rb]: note: `week_start_day` is called from `membership` - no longer the `group`

File: spec/models/activity_spec.rb [hunk 5]
```diff
@@ -107,11 +107,28 @@ def sleeping(attributes = {})
...
+          end
+        end
+
+        describe "when ending in the future" do
+          it "doesn't include this activity in the scope" do
+            expect do
+              sleeping(
+                start_time: start_time,
+                end_time: Time.current.advance(hours: 1))
+            end.to change { Activity.in_the_past.count }.by(0)
```
Comment (ericcf) [inline spec/models/activity_spec.rb]: FYI, you can also write this as `.not_to change { Activity.in_the_past.count }`. It's fine this way, just a note.

**Comment by ericcf** [comment]
Nice painstaking work!

**Pull request merged**